Updates for duration (#900)
neo4j/docs-cypher: modules/ROOT/pages/functions/temporal/duration.adoc

duration([ {years, quarters, months, weeks, days, hours, minutes, seconds, milliseconds, microseconds, nanoseconds} ])
----

*Arguments:*

> The map with the `DURATION` components as numeric expressions.

Reviewer: ```suggestion
The map of the `DURATION` components as numeric expressions.
```

| The values of the parameters may be expressed as decimal fractions.
| The values of the parameters may be arbitrarily large.
| The values of the parameters may be negative.

> | After creation of the `DURATION` object, you can query it for integer values of its components, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Reviewer 2: ```suggestion
| The xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[components of `DURATION` objects] are individually accessible.
```

edit by rsill-neo4j: removed "see" from the suggestion

| If `instant1` has a time zone component and `instant2` does not, the time zone component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| If `instant1` has a date component and `instant2` does not, the date component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| Any difference smaller than a whole month is disregarded.

> | Get the `DURATION` total number of months by calling `.months` on it, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Reviewer 3: ```suggestion
| Get the total number of months in a `DURATION` by returning the `months` component.
For more information, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].
```

> | Call the months component of the returned `DURATION` object via `.months` to get the duration total number of months, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Reviewer 2: ```suggestion
| Get the `DURATION` total number of months by calling `.months` on it, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].
```
And other occurrences

Author: 3 occurrences

| If `instant1` has a time zone component and `instant2` does not, the time zone component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| If `instant1` has a date component and `instant2` does not, the date component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| Any difference smaller than a whole month is disregarded.

> | To get the actual number of months as an `INTEGER`, call the months component of the returned `DURATION` object via `.months`, see see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Reviewer 2: ```suggestion
| You can get the duration total number of months through the `.months` component (see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations]).
```
_Component_ is a misleading word everywhere in these pages, as it implies it is only _part_ of the duration.

_Actual_ is also questionable, as it implies it is a more _real_ view than the duration itself. It's not: it's just a different representation.

These apply to further instances below, for seconds, minutes, etc.

> | To get the actual number of months as an `INTEGER`, call the months component of the returned `DURATION` object via `.months`, see see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Reviewer: ```suggestion
| Call the months component of the returned `DURATION` object via `.months` to get the duration total number of months, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].
```
For clarity and readability and to follow the tone of the rest of the section (ie avoid using _you_).

edit by rsill-neo4j: added an article

| If `instant1` has a time zone component and `instant2` does not, the time zone component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| If `instant1` has a date component and `instant2` does not, the date component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| Any difference smaller than a whole day is disregarded.

> | Get the `DURATION` total number of days by calling `.days` on it, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Reviewer 3: ```suggestion
| Get the total number of days in a `DURATION` by returning the `days` component.
For more information, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].
```

| If `instant1` has a time zone component and `instant2` does not, the time zone component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| If `instant1` has a date component and `instant2` does not, the date component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| Any difference smaller than a whole day is disregarded.

> | To get the actual number of days as an `INTEGER`, call the days component of the returned `DURATION` object via `.days`, see see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Author: ```suggestion
| You can get the duration total number of days through the `.days` component (see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations]).
```

> | To get the actual number of days as an `INTEGER`, call the days component of the returned `DURATION` object via `.days`, see see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Reviewer: ```suggestion
| Call the days component of the returned `DURATION` object via `.days` to get the duration total number of days, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].
```

| If `instant1` has a time component and `instant2` does not, the time component of `instant2` is assumed to be midnight, and vice versa.
| If `instant1` has a time zone component and `instant2` does not, the time zone component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| If `instant1` has a date component and `instant2` does not, the date component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.

> | Get the `DURATION` total number of seconds by calling `.seconds` on it, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Reviewer 3: ```suggestion
| Get the total seconds of days in a `DURATION` by returning the `seconds` component.
For more information, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].
```

| If `instant1` has a time component and `instant2` does not, the time component of `instant2` is assumed to be midnight, and vice versa.
| If `instant1` has a time zone component and `instant2` does not, the time zone component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.
| If `instant1` has a date component and `instant2` does not, the date component of `instant2` is assumed to be the same as that of `instant1`, and vice versa.

> | To get the actual number of seconds as an `INTEGER`, call the seconds component of the returned `DURATION` object via `.seconds`, see see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Author: ```suggestion
| You can get the duration total number of seconds through the `.seconds` component (see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations]).
```

> | To get the actual number of seconds as an `INTEGER`, call the seconds component of the returned `DURATION` object via `.seconds`, see see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].

Reviewer: ```suggestion
| Call the seconds component of the returned `DURATION` object via `.seconds` to get the duration total number of seconds, see xref:values-and-types/temporal.adoc#cypher-temporal-accessing-components-durations[Components of durations].
```
